flake.nix

```diff
@@ -0,0 +1,73 @@
+{
+  description = "Apple Silicon support for NixOS";
+
+  inputs = {
+    nixpkgs = {
+      # https://hydra.nixos.org/jobset/mobile-nixos/unstable/evals
+      # these evals have a cross-compiled stdenv available
+      url = "github:nixos/nixpkgs/6dccdc458512abce8d19f74195bb20fdb067df50";
+    };
+
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      flake = false;
+    };
+  };
+
+  outputs = { self, ... }@inputs:
+    let
+      # build platforms supported for uboot in nixpkgs
+      systems = [ "aarch64-linux" "x86_64-linux" ]; # "i686-linux" omitted
+
+      forAllSystems = inputs.nixpkgs.lib.genAttrs systems;
+    in
+      {
+        overlays = rec {
+          apple-silicon-overlay = import ./apple-silicon-support/packages/overlay.nix;
+          default = apple-silicon-overlay;
+        };
+
+        nixosModules = rec {
+          apple-silicon-support = ./apple-silicon-support;
+          default = apple-silicon-support;
+        };
+
+        packages = forAllSystems (system:
+          let
+            pkgs = import inputs.nixpkgs {
+              crossSystem.system = "aarch64-linux";
+              localSystem.system = system;
+              overlays = [
+                (import inputs.rust-overlay)
+                self.overlays.default
+              ];
+            };
+          in {
+            inherit (pkgs) m1n1 uboot-asahi;
```

Comment:
The issue I had with outputting the kernel package was https://github.com/NixOS/nix/issues/4265, which also breaks `nix flake show`. Luckily we should be able to get rid of the IFD using pure nix, as I've mentioned below.

We could also output the installer config, but I think it's not as clear where the config should go in the output, especially since we need to account for different possible build platforms, so I wanted to avoid it if possible.